README.md

```diff
@@ -1,2 +1,401 @@
-# example-application
-Example out-of-tree application that is also a module
+![Zephyr Project logo](https://github.com/zephyrproject-rtos/zephyr/blob/v1.14-branch/doc/images/Zephyr-Project.png)
+
+
+# Module Overview
+
+The Zephyr Project is a scalable real-time operating system (RTOS) supporting
+multiple hardware architectures, optimized for resource constrained devices,
+and built with security in mind.
+
+The Zephyr build system supports code tree "plug-ins" called *modules*.
```

Comment:
I am not convinced about this definition. Can you just reuse the words we have in the module doc?

Reply:
> I am not convinced about this definition. Can you just reuse the words we have in the module doc?

I agree with this statement in particular.

More generally, I really think that we should minimize the amount of documentation in this repository.

@gregshue while I understand and applaud the long term goal to have this integrate with Zephyr's documentation and CI systems, I honestly think you need to take a step back and realize that not even west has this level of integration. While you goals are worthy, this needs to start small, and I think avoiding documentation by linking to Zephyr's would be better for now.

Reply:
@mbolivar-nordic Thank you for the suggestion.  I have taken several steps back, forward, sideways, and up the ladder. It is hard to start smaller than noting functionality that already exists. :-)

When the Zephyr documentation generation system runs against `zephyrproject-rtos/zephyr/doc/`, it **already** pulls in documentation found in headers within this module's 'include/' folder. IIRC, it also generates documentation for the module's Kconfig variables. I presume it should (does?) do the same for at least board and sample documentation.

Whether the above behavior is correct or not, the Zephyr CI should verify that the Zephyr build and documentation generation systems do the right thing when valid document files are present in any module.

Reply:
@mbolivar-nordic The `west` reference actually illustrates my point.

-  `west` is independent of Zephyr, and Zephyr is where `west` extensions are implemented and usage is documented.

- `zephyr` must be **independent** of `example_application`, so `example_application` is where `zephyr` extensions should be implemented and usage should be documented.

The only question is documentation format and structure for modules owned and managed by the Zephyr Project. The amount of documentation does not matter.  Using anything other than the Zephyr structure and format for modules that could end up being submitted and added to Zephyr's default manifest or used for an extension example is counter-productive. This does not prevent others from using a different documentation system outside of the Zephyr Project organization.